app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,67 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isVisible/isUsed to true, while hiding the previous
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Make all terminals invisible (Becuase we don't have a pointer to the terminal that is currently visible)
+      _.each($scope.containerTerminals, function(term) {
+        term.isVisible = false;
+      });
+
+      newTerm.isVisible = true;
+      newTerm.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(existingTerminals) {
+      existingTerminals = existingTerminals || [];
+
+      // Remake list with old list's isVisible/isUsed data
+      _.each($scope.pod.spec.containers, function(spec) {
```

Comment:
prob should be ` _.each($scope.pod.spec.containers, function(container)`, since there is one spec that contains multiple containers.